juju/machine.py

```diff
@@ -176,15 +176,15 @@ def run(self, command, timeout=None):
         """ Execute an scp command. Requires a fully qualified source and
         destination.
         """
-        cmd = [
-            'scp',
+        cmd = ['scp']
+        cmd += scp_opts.split()
+        cmd.extend([
             '-i', os.path.expanduser('~/.local/share/juju/ssh/juju_id_rsa'),
             '-o', 'StrictHostKeyChecking=no',
             '-q',
             '-B',
             source, destination
-        ]
-        cmd += scp_opts.split()
+        ])
```

Comment:
I think the original intention of add the `scp_opts` at the end was to allow user-supplied options to override the defaults.  I'm not entirely certain if that would work as expected, but maybe it would be best to inject the user options between the defaults and `source` / `destination` instead of either at the beginning or end.  Something like:

```python
        cmd = [
            '-i', os.path.expanduser('~/.local/share/juju/ssh/juju_id_rsa'),
            '-o', 'StrictHostKeyChecking=no',
            '-q',
            '-B',
        ]
        cmd.extend(scp_opts.split() if isinstance(scp_opts, str) else scp_opts)
        cmd.extend([source, destination])
```